airflow/providers/amazon/aws/operators/glue.py

```diff
@@ -101,7 +101,8 @@ def execute(self, context):
         if self.script_location and not self.script_location.startswith(self.s3_protocol):
             s3_hook = S3Hook(aws_conn_id=self.aws_conn_id)
             script_name = os.path.basename(self.script_location)
-            s3_hook.load_file(self.script_location, self.s3_bucket, self.s3_artifacts_prefix + script_name)
+            s3_hook.load_file(self.script_location, self.s3_artifacts_prefix + script_name, bucket_name=self.s3_bucket)
+            self.script_location = f"s3://{self.s3_bucket}/{self.s3_artifacts_prefix + script_name}"
```

Comment:
This value should be resolved in `__init__()` instead.

Reply:
Or this should be a local variable in this function. The important thing is `execute()` should not modify attributes.

Reply:
I have resolved this in the latest push